### ISSUE TITLE
So if I define ISSUE_SUPPORT and recompile, issue content will be displayed?

### ISSUE THREAD

**Issue by Jibun-no-Kage** [util-linux/util-linux#1667]
So if I define ISSUE_SUPPORT and recompile, issue content will be displayed?  Do I understand this correctly?  We have a special use case where we need to display a legal policy statement (i.e. a few lines of text) before ANY prompt or operator action.  Seems like enable of ISSUE_SUPPORT would allow this.  Maybe even change the code, so  that /etc/issue.legal or such unique named file would be displayed to explicitly do what we need in our use case.  Rather than defaulting to /etc/issue or /etc/issue.net.  Please advise.  Of course this needs to be done against all consoles, serial, virtual, etc.

**Comment by karelzak** [comment]
It should be automatically defined if LOGIN_PROCESS or SYSV_STYLE macros exist. And I guess it's enabled almost everywhere, so you do not need to define anything.

So, the question is why do you ask? ;-) Do you have any problem seeing /etc/issue?

I don't think any local change to the code for your purpose is a good idea, it's usually an expensive solution to maintain non-upstream changes, just put your stuff to /etc/issue (or define another path by --issue-file).

**Comment by Jibun-no-Kage** [comment]
Actually I am working to establish a 'legal' banner that is displayed on every console before any operator id is entered.  This is a security as well as a legal requirement, where the warning must be displayed immediately, but with a typical serial or virtual consoles, at least enter must be sent to 'open' said console.  Fortunately, We can do this Microsoft Windows, and even with GNOME greeter configuration.  But serial consoles, this is not done as required by default, only after user id is enter is any banner text displayed via issue or issue.net, this is the PuTTY designed behavior, for example, but is not acceptable to our legal department, or our security team.  They want the prosecution disclaimer on serial consoles as well as virtual terminal sessions at terminal session open.  The most logical way to do this seems to display the text before or as part of the actual 'login as' text is displayed.  If we could customize the 'login as' prompt that would do nicely.

For example...

"The following system is monitored, and access is explicitly controlled.  Disconnect now unless you have authorized access to this system, or face legal action, but to and including termination, to the full extent that the law allows.  This is the only warning you will receive.<NEWLINE><NEWLINE> Login as:"

Again the above must be displayed at the absolute soonest possible point, which per our legal department is before the 'login as' prompt is displayed.

**Comment by Jibun-no-Kage** [comment]
To clarify the goal, issue and issue.net are displayed ONLY after the first login, and logout sequence are completed, after that issue or issue.net is displayed.  But this is incomplete, issue or issue.net should be, to be consistent, be displayed before 'login as' is presented at the initial console session open/start.

If I open serial console or PuTTY... and hit return, 'login as' is displayed but not issue or issue.net.  After logout, then issue or issue.net is displayed.  This is very odd behavior.  As you can see from the example below.

-----

raspberrypi login: root
Password:
Linux raspberrypi 5.15.32-v7+ #1538 SMP Thu Mar 31 19:38:48 BST 2022 armv7l

The programs included with the Debian GNU/Linux system are free software;
the exact distribution terms for each program are described in the
individual files in /usr/share/doc/*/copyright.

Debian GNU/Linux comes with ABSOLUTELY NO WARRANTY, to the extent
permitted by applicable law.
Last login: Tue May  3 04:06:25 HDT 2022 on ttyAMA0
root@raspberrypi:~# logout

[ ISSUE/ISSUE.NET TEXT ]
Raspbian GNU/Linux 11 raspberrypi ttyAMA0

raspberrypi login:

-----

GeTTY service has the same odd behavior, the issue content is ONLY displayed after logout, so the 2nd session gets the text banner but the 1st session NEVER gets the text banner.  Even if I create a GeTTY override and create a custom text banner it is displayed after logout of session, not at open of initial session for the given (virtual) console.

**Comment by Jibun-no-Kage** [comment]
Just as a test to understand the complexity of changing the login executable, I attempted the following...

# apt install build-essential git

# git clone https://github.com/util-linux/util-linux.git

# apt install apt-file
# apt-file update

# cd util-linux
# ./autogen.sh
# apt install autoconf autopoint bison gettext libtool libuser
# ./autogen.sh
# ./configure --disable-all-programs --enable-login

But I hit a error I am somewhat confused by...

checking for libuser >= 0.58... no
configure: error: login selected, but required PAM header file not available

The library libuser is present...

# dpkg -l | grep libuser
ii  libuser                              1:0.62~dfsg-0.4                  armhf        user and group account administration library - utilities

**Comment by karelzak** [comment]
agetty displays issue file(s) before login and no interaction with the user is necessary. Note that agetty does not have to be in the game in some ways how to log in to the system, for example, it's not used for ssh sessions, telnets, etc.

After log in the system usually displays /etc/motd (or /etc/motd.d).

Ad login compilation, for example, ubuntu uses libpam-dev ...

**Comment by Jibun-no-Kage** [comment]
@karelzak  Thanks for the information, last night I figured out which which header file was missing, and using apt-file, found the right package, so I was able to compile both login and agetty, just as  learning test for any potential customization we might need.

As for getty service, i.e. agetty, as implemented, yes I was able to change the override.conf to display the issue file, or a customized file as needed.  What was interesting was it seemed like a reboot was required to get getty service to show the issue file as expected.  Before reboot could not get the issue file to display initially at the very first console session open.  It would display after logout... at the next session open.  As long as we trigger a reboot after configuration/deployment, serial (UART) console banner is displayed before operator user id input, which is what we needed.   I have yet to test virtual serial console function but it should be comparable.

Tthe first thing our security team noted was that sub shelling does not display the same issue file text as initial login... so a local sub-shell or even a remote ssh connection never gets the banner text like via agetty issue method.

PuTTY its self is a challenge of course because any banner text is displayed, after user id entered but before password entered.  But that is different challenge of course.

Good news is at least for the initial console sessions, point of entry, what is supported works, similar to how the UI greeter banner text  works.  I have challenged the security team to accept that as long as the initial sessions have the banner warning, it meets the policy requirement, IMHO.  If a user creates a ssh session or sub shell, they still had to acknowledge the 'policy' is in effect at the initial login at console.

Thanks for the input and clarifications.  Greatly appreciated.
